packages/blockchain-wallet-v4-frontend/src/data/middleware/webSocket/socketd/sagas.js

```diff
@@ -0,0 +1,284 @@
+import { call, select, put } from 'redux-saga/effects'
+import { concat, equals, prop } from 'ramda'
+import { selectors } from 'data'
+import * as actions from '../../../actions'
+import * as T from 'services/AlertService'
+import { WALLET_TX_SEARCH } from '../../../form/model'
+import {
+  header,
+  ethSentConfirmed,
+  ethReceivedPending,
+  ethReceivedConfirmed,
+  bitcoinTransaction
+} from './messageTypes'
+
+export default ({ api, socket }) => {
+  const send = socket.send.bind(socket)
+
+  const onOpen = function * () {
+    try {
+      // 1. subscribe to block headers
+      yield call(
+        send,
+        JSON.stringify({ command: 'subscribe', entity: 'header', coin: 'btc' })
+      )
+      yield call(
+        send,
+        JSON.stringify({ command: 'subscribe', entity: 'header', coin: 'bch' })
+      )
+      yield call(
+        send,
+        JSON.stringify({ command: 'subscribe', entity: 'header', coin: 'eth' })
+      )
+
+      /* wallet */
+      let walletContext = yield select(
+        selectors.core.wallet.getInitialSocketContext
+      )
+
+      // 2. subscribe to accounts
+      walletContext.addresses.forEach(address => {
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'account',
+            coin: 'btc',
+            param: { address }
+          })
+        )
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'account',
+            coin: 'bch',
+            param: { address }
+          })
+        )
+      })
+
+      // 3. subscribe to xpubs
+      walletContext.xpubs.forEach(xpub => {
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'xpub',
+            coin: 'btc',
+            param: { address: xpub }
+          })
+        )
+        send(
+          JSON.stringify({
+            command: 'subscribe',
+            entity: 'xpub',
+            coin: 'bch',
+            param: { address: xpub }
+          })
+        )
+      })
+
+      // 4. subscribe to lockbox xpubs
+      const { data: btcLockboxContext } = yield select(
```

Comment:
never access the `data` property of a `Remote` type directly.  It always needs to accessed by either mapping over the `Remote` or by calling the `getOrElse()` or `getOrFail()` prototype functions.  In this specific case if we fail to get Locbkox context we dont want to error and instead just provide an empty fallback.  So this should be:
```
const btcLockboxContext =  (yield select(selectors.core.kvStore.lockbox.getLockboxBchContext).getOrElse([])
```